tests/backup/backup_helper.go

```diff
@@ -1181,13 +1247,128 @@ func backupSuccessCheck(backupName string, orgID string, retryDuration int, retr
 
 	_, err := task.DoRetryWithTimeout(backupSuccessCheck, time.Duration(retryDuration)*time.Minute, time.Duration(retryInterval)*time.Second)
 	if err != nil {
-		return false, err
+		return false, nil, err
 	}
-	return true, nil
+	backupInspectRequest := &api.BackupInspectRequest{
+		Name:  backupName,
+		Uid:   bkpUid,
+		OrgId: orgID,
+	}
+	resp, err := backupDriver.InspectBackup(ctx, backupInspectRequest)
+	if err != nil {
+		return false, nil, err
+	}
+	backupStatus := (resp.GetBackup().GetStatus().Status == api.BackupInfo_StatusInfo_Success)
+	log.Infof("Backup [%s] created successfully", backupName)
+	return backupStatus, resp, nil
+}
+
+func ValidateBackup(backupInspectResponse *api.BackupInspectResponse, clusterAppsContexts []*scheduler.Context) (*BackupRestoreContext, error) {
+	bkpObjs := backupInspectResponse.Backup.Resources
+	if backupSpecContext, err := GetBackupSpecObjectsContexts(bkpObjs, clusterAppsContexts); err != nil {
+		return nil, fmt.Errorf("GetBackupSpecObjects Err: %v", err)
+	} else {
+		return &BackupRestoreContext{
+			schedCtxs: backupSpecContext,
+		}, nil
+	}
+}
+
+// Filters the source namespaces' contexts to only include backup objects
+// MAKE SURE to include Contexts of ALL namespaces which contain the backup objects
+func GetBackupSpecObjectsContexts(resourceInfos []*api.ResourceInfo, clusterAppsContexts []*scheduler.Context) ([]*scheduler.Context, error) {
+	log.InfoD("Validation: Getting the backup objects (specs) from the Context")
+
+	var backupAppContexts []*scheduler.Context = make([]*scheduler.Context, 0)
+	var allBkpObjs []*api.ResourceInfo = make([]*api.ResourceInfo, 0)
+
+	//for each clusterAppsContext (namespace), we return the corresponding BackupSpecObject
+	for _, clusterAppsContext := range clusterAppsContexts {
+		var specObjects []interface{} = make([]interface{}, 0)
+		// the Backup objects which were found in this namespace
+		var resInfoObjs []*api.ResourceInfo = make([]*api.ResourceInfo, 0)
+
+		for _, resource := range resourceInfos {
+			if resource.Namespace == clusterAppsContext.ScheduleOptions.Namespace {
+				resInfoObjs = append(resInfoObjs, resource)
+			} else if resource.Namespace == "" {
+				// IF it is therein this context, we'll work with it
+				resInfoObjs = append(resInfoObjs, resource)
+			}
+		}
+
+	specloop:
+		for _, spec := range clusterAppsContext.App.SpecList {
+			name, kind, ns, err := GetSpecNameKindNamepace(spec)
+			if err != nil {
+				log.Error(err)
```

Comment:
why you want to continue if there is an error

Reply:
That's because `GetSpecNameKindNamepace` only errors out if the object whose properties I'm trying to obtain is unsupported by the function. This happens, for example, in case of CRs (like `kind: elastic-search`). This only affects the "cloning" of _that particular object_, and not the rest of the process. It can be **safely ignored in most cases** where cloning that particular object **doesn't really matter**, but if it _does_ matter to someone, then the error will tell them that they need to care about it.

Technically speaking, the only case this occurs right now, is for CRs, for which I didn't write the necessary function, since I was focusing on getting the test working and that extra functionality isn't required at this moment, but it will cause an error (which can be ignored). I'll file  bug for this and this can be removed when the bug is resolved. At the moment, this is definitely a P3 or P4 bug, so can be ignored.

can this be considered resolved for now?